sdk/face/ai-vision-face-rest/samples-dev/findSimilars.ts

```diff
@@ -0,0 +1,137 @@
+import { readFileSync } from 'fs';
+import { AzureKeyCredential } from '@azure/core-auth';
+
+import createFaceClient, {
+    AddLargeFaceListFace200Response,
+    CreateLargeFaceList200Response,
+    DeleteLargeFaceList200Response,
+    Detect200Response,
+    FindSimilar200Response,
+    FindSimilarFromLargeFaceList200Response,
+    TrainLargeFaceList202Response,
+    getLongRunningPoller,
+} from '@azure-rest/ai-vision-face';
+
+/**
+ * This sample demonstrates how to create a liveness detection session.
+ *
+ * @summary creates a liveness detection session
+ */
+
+async function findSimilarsFromFaceIds() {
+    const endpoint = process.env['FACE_ENDPOINT'] ?? '<endpoint>';
+    const apikey = process.env['FACE_APIKEY'] ?? '<apikey>';
+    const credential = new AzureKeyCredential(apikey);
+    const client = createFaceClient(endpoint, credential);
+
+    const detect = async (fileName: string): Promise<string[]> => {
+        const response = await client.path('/detect').post({
+            contentType: 'application/octet-stream',
+            queryParameters: {
+                detectionModel: 'detection_03',
+                recognitionModel: 'recognition_04',
+                returnFaceId: true,
+            },
+            body: readFileSync(`samples-dev/data/${fileName}`),
+        }) as Detect200Response;
```

Comment:
why do we need casting here ?

Reply:
Otherwise, TypeScript compiler won't recognize the type of `response.body` and the statement `response.body.map` will cause compilation error.

Reply:
you should leverage the isUnexpected helper like this one 
https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/contentsafety/ai-content-safety-rest/test/public/contentSafety.spec.ts#L50-L52 which should help you narrow the type, and you don't need casting.

Reply:
This solution looks good, but I'm not sure if it's applicable to our SDK. Please refer to our [clientDefinitions.ts](https://github.com/chungshengfu/azure-sdk-for-js/blob/user/chfu/ai-vision-face/sdk/face/ai-vision-face-rest/src/clientDefinitions.ts). We have so many return types so there will be tons of overloads of `isUnexpected`.

Reply:
I just notice it's generated code. Let go this way.